test/Microsoft.TestPlatform.TestHostProvider.UnitTests/Hosting/TestHostManagerCallbacksTests.cs

```diff
@@ -94,4 +98,103 @@ public void ErrorReceivedCallbackShouldAppendNewLineApproprioritlyWhenReachingMa
 
         Assert.AreEqual("1234" + Environment.NewLine.Substring(0, 1), _testHostProcessStdError.ToString());
     }
+
+    [TestMethod]
+    public void ErrorReceivedCallbackShouldNotCrashIfInvalidProcessHandle()
+    {
+        var handleProcessHelper = new InvalidHandleProcessHelper();
+        bool onHostExitedCalled = false;
+        TestHostManagerCallbacks.ExitCallBack(handleProcessHelper, null, new StringBuilder(),
+        hostProviderEventArgs =>
+        {
+            onHostExitedCalled = true;
+            Assert.AreEqual(-1, hostProviderEventArgs.ErrroCode);
+        });
```

Comment:
Nit: fix indentation.

```suggestion
        TestHostManagerCallbacks.ExitCallBack(handleProcessHelper, null, new StringBuilder(),
            hostProviderEventArgs =>
            {
                onHostExitedCalled = true;
                Assert.AreEqual(-1, hostProviderEventArgs.ErrroCode);
            });
```